server/server.go

```diff
@@ -272,7 +322,11 @@ func (r *rodeServer) CreatePolicy(ctx context.Context, policyEntity *pb.PolicyEn
 		return nil, status.Errorf(codes.InvalidArgument, "policy name not provided")
 	}
 
-	// CheckPolicy call will go here
+	// CheckPolicy before writing to elastic
+	result, err := r.ValidatePolicy(ctx, &pb.ValidatePolicyRequest{Policy: policyEntity.RegoContent})
+	if (err != nil) || !result.Compile {
+		return nil, createError(log, "failed to compile the provided policy", err)
```

Comment:
Just going to leave this comment as a reminder - can we return the same type of response that is returned from the validate policy endpoint here?